### PR TITLE
refactor: Convert CollectionHelper to Kotlin

### PR DESCRIPTION
**Pull request by david-allison**
The biggest change left

* Removed LazyHolder and used `companion object` for `.instance`

### CODE REVIEW

**Comment by david-allison** [comment]
https://github.com/ankidroid/Anki-Android/pull/12217 merged, rebased on `main` + ready for review

**Comment by BrayanDSO** [comment]
```
/home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/CollectionHelper.kt:1: Error: Tags were corrupted by the Kotlin migration. For example: <br></br> [KotlinMigrationFixLineBreaks from com.ichi2.anki:lint-rules]
/***************************************************************************************
```

```
e: /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/test/java/com/ichi2/anki/AbstractFlashcardViewerTest.kt: (180, 32): Unresolved reference: runTest
e: /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/test/java/com/ichi2/anki/AbstractFlashcardViewerTest.kt: (224, 40): Unresolved reference: runTest
e: /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/test/java/com/ichi2/anki/CardBrowserTest.kt: (356, 45): Unresolved reference: runTest
e: /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/test/java/com/ichi2/anki/CardBrowserTest.kt: (369, 42): Unresolved reference: runTest
e: /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/test/java/com/ichi2/anki/CardBrowserTest.kt: (583, 31): Unresolved reference: runTest
e: /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/test/java/com/ichi2/anki/CardBrowserTest.kt: (641, 40): Unresolved reference: runTest
e: /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/test/java/com/ichi2/anki/ReviewerNoParamTest.kt: (147, 40): Unresolved reference: runTest
e: /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/test/java/com/ichi2/anki/ReviewerTest.kt: (159, 31): Unresolved reference: runTest
e: /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/test/java/com/ichi2/anki/ReviewerTest.kt: (179, 9): Suspend function 'undo' should be called only from a coroutine or another suspend function
e: /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/test/java/com/ichi2/anki/ReviewerTest.kt: (188, 35): Unresolved reference: runTest
e: /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/test/java/com/ichi2/anki/ReviewerTest.kt: (212, 9): Suspend function 'undo' should be called only from a coroutine or another suspend function
```

+ conflicts

**Comment by dae** [comment]
The rebase has reverted the runTest/awaitJob changes from #12226

File: AnkiDroid/src/main/java/com/ichi2/anki/CollectionHelper.kt [hunk 1]
```diff
@@ -183,379 +120,444 @@ public class CollectionHelper {
...
-
-            Long maybeCurrentCollectionSizeInBytes = getCollectionSize(context);
-            if (maybeCurrentCollectionSizeInBytes == null) {
-                Timber.w("Error obtaining collection file size.");
-                String requiredFreeSpace = defaultRequiredFreeSpace(context);
-                return fromError(context.getResources().getString(R.string.integrity_check_insufficient_space, requiredFreeSpace));
+        /**
+         * Create the AnkiDroid directory if it doesn't exist and add a .nomedia file to it if needed.
+         *
+         * The AnkiDroid directory is a user preference stored under [.PREF_COLLECTION_PATH], and a sensible
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/anki/CollectionHelper.kt:335]:
```suggestion
         * The AnkiDroid directory is a user preference stored under [PREF_COLLECTION_PATH], and a sensible
```

**Review by BrayanDSO — CHANGES_REQUESTED** [review]
For the future, I'd consider using [object](https://kotlinlang.org/docs/object-declarations.html#object-declarations-overview) on CollectionHelper. It would reduce the boilerplace of writing `.instance` and make possible to import/use the methods directly.

Only `setInstanceForTesting` would need changes to adapt to it

File: AnkiDroid/src/main/java/com/ichi2/anki/CollectionHelper.kt [hunk 2]
```diff
@@ -14,167 +14,104 @@
...
+    open fun getCol(context: Context?): Collection? {
+        return getColUnsafe()
     }
 
     /**
-     * Given a path to a .anki2 file returns an open {@link Collection} associated with the path.
+     * Given a path to a .anki2 file returns an open [Collection] associated with the path.
      *
-     * This operation does not call {@link #initializeAnkiDroidDirectory} and does not set the singleton instance's {@link #mCollection}
+     * This operation does not call [.initializeAnkiDroidDirectory] and does not set the singleton instance's [.mCollection]
```
[the remark this answers is not in the document]
Comment (BrayanDSO) [reply AnkiDroid/src/main/java/com/ichi2/anki/CollectionHelper.kt]: The dot before mCollection wasn't removed

**Review by BrayanDSO — APPROVED** [review]
LGTM, not worth blocking on a comment

File: AnkiDroid/src/main/java/com/ichi2/anki/CollectionHelper.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/anki/CollectionHelper.kt]: `mCollection` isn't a valid symbol, I've done it, but it'll be changed in a later cleanup

**Review by lukstbit — APPROVED** [review]
LGTM

**Pull request merged**